src/edtr-io/plugins/image.tsx

```diff
@@ -85,44 +93,102 @@ export function createUploadImageHandler(getCsrfToken: () => string) {
   }
 }
 
-export function createReadFile(getCsrfToken: () => string) {
+export function createReadFile() {
   return function readFile(file: File): Promise<LoadedFile> {
     return new Promise((resolve, reject) => {
-      const reader = new FileReader()
-
-      reader.onload = function (e: ProgressEvent) {
-        if (!e.target) return
-        const formData = new FormData()
-        formData.append('attachment[file]', file)
-        formData.append('type', 'file')
-        formData.append('csrf', getCsrfToken())
-
-        fetch('/attachment/upload', {
-          method: 'POST',
-          body: formData,
-        })
-          .then((response) => response.json())
-          .then((data: { success: boolean; files: { location: string }[] }) => {
-            if (!data['success']) reject()
-            resolve({
-              file,
-              dataUrl: data.files[0].location,
-            })
-          })
-          .catch(() => {
-            reject()
+      const authenticationPayload = parseAuthCookie()
+
+      if (!authenticationPayload?.token) return
+
+      const gqlFetch = createAuthAwareGraphqlFetch({
+        current: authenticationPayload,
+      })
+      const args = JSON.stringify({
+        query: uploadUrlQuery,
+        variables: { mediaType: MediaType.ImagePng },
```

Comment:
This must match the content type given later in https://github.com/serlo/frontend/blob/9d8c02a9ba5c2bb93726ac5625dc22b528040a4b/src/edtr-io/plugins/image.tsx#L123 -> This is the reason why currently we get an error.

Reply:
🤦‍♂️